fixes/consolidate-multiple-richtext-history.sql

```diff
@@ -0,0 +1,157 @@
+/*
+ *	Consolidate field history records, on configurable time threshold (@timeThreshold) in minutes.
+ *
+ *	NOTE:  This script defaults to rolling back changes.
+ *		To commit changes, set @saveChanges = 1.
+ */
+declare @saveChanges bit; --set @saveChanges = 1
+declare @TableName varchar(100); --set @TableName = 'Test'; --Table Name Ex: Test, RegressionTest, BaseAsset
+declare @Table varchar(100); set @Table = quotename(@TableName);
+declare @Table_Now varchar(100); set @Table_Now = quotename(@TableName + '_Now');
+declare @FieldName varchar(100); --set @FieldName = 'ExpectedResults'; -- Table attribute Ex: Description, ExpectedResults, Setup
+declare @Field varchar(100); set @Field = quotename(@FieldName);
+declare @timeThreshold int;
+
+set @timeThreshold = 5
+
+create table #assets(ID int not null, CurrentAuditID int not null, NextAuditID int not null)
+create table #suspect(ID int not null, CurrentAuditID int not null, NextAuditID int not null)
+create table #bad(ID int not null, CurrentAuditID int not null, NextAuditID int not null)
+
+set nocount on; begin tran; save tran TX
+
+-- Asset Rows
+declare @q1 varchar(max);
+select @q1 = ';
+with H as (
+	select AA.ID, AA.AssetType, AuditID, R=ROW_NUMBER() OVER(partition by AA.ID, AA.AssetType order by AuditID)
+	from dbo.AssetAudit AA
+)
+insert #assets(ID, CurrentAuditID, NextAuditID)
+select aH.ID, aH.AuditID, bH.AuditID
+from H as aH
+join dbo.' + @Table + ' aAsset on aAsset.ID=aH.ID and aAsset.AssetType=aH.AssetType and aH.AuditID = aAsset.AuditBegin
+join H as bH on bH.ID=aH.ID and aH.AssetType=bH.AssetType and bH.R=aH.R+1
+join dbo.' + @Table + ' bAsset on bAsset.ID=bH.ID and bAsset.AssetType=bH.AssetType and bH.AuditID = bAsset.AuditBegin'
+
+-- Consecutive Asset changes
+declare @q2 varchar(max);
+select @q2 = ';
+insert #suspect(ID, CurrentAuditID, NextAuditID)
+select Assets.ID, Assets.CurrentAuditID, Assets.NextAuditID
+from #assets as Assets
+join dbo.' + @Table + ' currentAsset on currentAsset.ID=Assets.ID and currentAsset.AuditBegin=Assets.CurrentAuditID
+join dbo.' + @Table + ' nextAsset on nextAsset.ID=Assets.ID and nextAsset.AuditBegin=Assets.NextAuditID
+join dbo.Audit currentA on currentA.ID = Assets.CurrentAuditID
+join dbo.Audit nextA on nextA.ID = Assets.NextAuditID
+WHERE
+ISNULL(currentA.ChangedByID,-1) = ISNULL(nextA.ChangedByID,-1)  -- Consecutive changes from the same user
+AND DATEDIFF(mi,currentA.ChangeDateUTC,nextA.ChangeDateUTC) <= ' + CAST(@timethreshold as varchar(10)) + ' --Time threshold / period / lapse.
+AND	ISNULL(currentAsset.' + @Field + ',-1) != ISNULL(nextAsset.' + @Field + ',-1) --@FieldName has changed '
+
+-- Asset column comparison
+declare @colsAB varchar(max)
+select @colsAB=(
+	select REPLACE(' and (A.{col}=C.{col} or (A.{col} is null and C.{col} is null))', '{col}', quotename(COLUMN_NAME))
+	from INFORMATION_SCHEMA.COLUMNS C
```

Comment:
Aliasing this join `C` is pretty confusing, given the use of `C.` in the output.  Could we just not alias it at all?

Reply:
`C` is the "starting value" join, and all three names (`A` for "current", `B` for "next", and `C`) could use better names that actually say what they are.

Reply:
Heh, I was reserving comment on the name `@colsAB`.  B?